### DISCUSSION TITLE
Positional feedback

### DISCUSSION THREAD

**Discussion by dwgong** [slmsuite/slmsuite#58] [Q&A] [answered]
Dear developer, is it true that the current package only takes care of the intensity for each spot during camera feedback but not the position? Do you have an intention to implement such function? Thanks for your time! I would be happy to discuss this idea with you since our group is also thinking about it.

**Comment by ichristen** [comment, marked as answer]
Hi @dwgong ,

This is already implemented as [`SpotHologram.refine_offset()`](https://slmsuite.readthedocs.io/en/latest/_autosummary/slmsuite.holography.algorithms.SpotHologram.html#slmsuite.holography.algorithms.SpotHologram.refine_offset). Usually this needs a few iterations to hone the positions, so you can either call it between instances of the `optimize()` loop or as part of a `callback=` function to `optimize()`. Let us know if this is what you're looking for! You're correct that this function is a bit hidden, so we'll add it to the examples in the next release.

Best,
Ian

**Reply by ichristen** [reply]
(This is not enabled by default in the way that intensity weighting is because of the chance of unusual behavior, esp. if the spots are reflected off of patterned substrates.)

**Reply by dwgong** [reply]
lol I see, thanks a lot! I'll try and play with it.
